src/weekdays.rs

```diff
@@ -0,0 +1,723 @@
+        self.days.remove(next_back);
+        Some(next_back)
+    }
+}
+impl ExactSizeIterator for WeekdaysIterFrom {
+    fn len(&self) -> usize {
+        self.days.len().into()
+    }
+}
+impl FusedIterator for WeekdaysIterFrom {}
+
+impl fmt::Display for Weekdays {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        let mut iter = self.iter();
+        if let Some(first) = iter.next() {
+            write!(f, "{first}")?;
+            for weekday in iter {
+                write!(f, " | {weekday}")?;
+            }
+            Ok(())
+        } else {
+            write!(f, "<empty>")
+        }
+    }
+}
+
+// impl Bit* for Weekdays
+impl BitOr for Weekdays {
+    type Output = Self;
+
+    fn bitor(self, rhs: Self) -> Self::Output {
+        self.union(rhs)
+    }
+}
+impl BitAnd for Weekdays {
+    type Output = Self;
+
+    fn bitand(self, rhs: Self) -> Self::Output {
+        self.intersection(rhs)
+    }
+}
+impl BitXor for Weekdays {
+    type Output = Self;
+
+    fn bitxor(self, rhs: Self) -> Self::Output {
+        self.symmetric_difference(rhs)
+    }
+}
+#[cfg(test)]
+#[test]
+fn bitwise_set_operations_preserve_8th_bit_invariant() {
+    for set1 in Weekdays::iter_all() {
+        for set2 in Weekdays::iter_all() {
+            (set1 | set2).assert_8th_bit_invariant();
+            (set1 & set2).assert_8th_bit_invariant();
+            (set1 ^ set2).assert_8th_bit_invariant();
+        }
+    }
+}
+
+// impl Bit*Assign for Weekdays
+impl BitOrAssign for Weekdays {
+    fn bitor_assign(&mut self, rhs: Self) {
+        self.0 |= rhs.0;
+    }
+}
+impl BitAndAssign for Weekdays {
+    fn bitand_assign(&mut self, rhs: Self) {
+        self.0 &= rhs.0;
+    }
+}
+impl BitXorAssign for Weekdays {
+    fn bitxor_assign(&mut self, rhs: Self) {
+        self.0 ^= rhs.0;
+    }
+}
+
+impl Not for Weekdays {
+    type Output = Self;
+
+    fn not(self) -> Self::Output {
+        self.inverse()
+    }
+}
+#[cfg(test)]
+#[test]
+fn not_operation_preserves_8th_bit_invariant() {
+    for weekdays in Weekdays::iter_all() {
+        (!weekdays).assert_8th_bit_invariant();
+    }
+}
+
+impl From<Weekday> for Weekdays {
+    fn from(weekday: Weekday) -> Self {
+        Self::from_one(weekday)
+    }
+}
+
+impl Extend<Weekday> for Weekdays {
+    fn extend<T: IntoIterator<Item = Weekday>>(&mut self, iter: T) {
+        for weekday in iter {
+            self.insert(weekday);
+        }
+    }
+}
+impl FromIterator<Weekday> for Weekdays {
+    fn from_iter<T: IntoIterator<Item = Weekday>>(iter: T) -> Self {
+        let mut weekdays = Self::EMPTY;
+        weekdays.extend(iter);
+        weekdays
+    }
+}
+
+impl IntoIterator for Weekdays {
+    type Item = Weekday;
+    type IntoIter = WeekdaysIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.iter()
+    }
+}
+
+// impl Bit*<Weekday> for Weekdays
+impl BitOr<Weekday> for Weekdays {
```

Comment:
This set of operations might be unnecessary. `days | day` looks better than `days | Weekdays::from(day)` and the meaning seems clear to me, but I may be biased.

Reply:
Let's drop them for now.